src/dict.h

```diff
@@ -44,7 +44,16 @@
 #define DICT_OK 0
 #define DICT_ERR 1
 
-typedef struct dictEntry dictEntry; /* opaque */
+typedef struct dictEntry {
+    void *key;
+    union {
+        void *val;
+        uint64_t u64;
+        int64_t s64;
+        double d;
+    } v;
+    struct dictEntry *next;     /* Next entry in the same hash bucket. */
+} dictEntry;
```

Comment:
#11465 moved the struct definition to dict.c and kept only `typedef struct dictEntry dictEntry; /* opaque */` here. Putting it back here is a merge-conflict mistake.

Reply:
I'm happy to revert it back, just want to understand why this decision has been made. Can you clarify motivation behind having opaque definition in the .h file? Does it allow multiple different concrete struct implementations at the same time or is it something else?

Reply:
Exactly. Then we can do refactoring and optimization within dict.c without affecting anything else.

Reply:
Done